### ISSUE TITLE
Block reassurance and multistore content

### ISSUE THREAD

**Issue by momo1240** [PrestaShop/PrestaShop#29219]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Hello 
I'm using the latest version of prestashop with the latest version of block reassurance module.
I noticed that we can't set different content in multistore configuration

thank you for your help

### Describe the solution you'd like

For example, disabling some blocks in a defined store and keeping them enabled in others 

### Alternatives you've considered

None

### Additional context

How to reproduce it : 

in Multi store configuration : 

1. go to block reassurance module
2. select a store and disable a block
3. when you check other stores the block is also disable

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @momo1240 

We're aware of it. There are some plans to improve this module, but there's no ETA as there are no resources right now to take on this module.

If you need this feature, you can contribute directly to this module in its repository:
https://github.com/PrestaShop/blockreassurance

**Comment by hibatallahAouadni** [comment]
Hello @kpodemski & @momo1240 

I think this issue is similar to this one #28446, it seems that the problem occurs not only with this module but all native module.
Could you please check it and feedback.

Thanks!

**Comment by kpodemski** [comment]
@hibatallahAouadni I don't think it is the same issue, `blockreassurance` is not compatible with the multi-store, it is not about the cache :)

**Comment by AureRita** [comment]
Hi @momo1240 , 

As @kpodemski said, `blockreassurance` isn't compatible with the multi-store, so do you want to transform this issue into a new feature ?

waiting for your feedback

**Comment by momo1240** [comment]
Hello @AureRita,
yes it would be great
thank you

**Comment by AureRita** [comment]
hi @momo1240, 

ok, I'll change the description of your "issue" tell me if this new description suits you 

Waiting for your feedback

**Comment by AureRita** [comment]
I changed your description, thank you for your suggestion. The Maintainer Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you create it. 

If you or anyone else is willing to [contribute this feature](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/) to the PrestaShop Project, the teams will be happy to help by ensuring that it fits well within the project. Please let us know if that is the case!

Thank you

**Comment by Triloworld** [comment]
This issue is to refactor this lines:
https://github.com/PrestaShop/blockreassurance/blob/29bf69904b1e61a5375e86860c26af421683fcd8/classes/ReassuranceActivity.php#L144-L168

SQL query need to be adjusted to current ID shop which is stored in database.
When enabling multistore - we get blocks from all shops and same language in all view where we using it.

**Comment by boo-code** [comment]
Context this thread is missing: the module had multistore support and it was deliberately removed four months before this issue was opened.

PrestaShop/blockreassurance#336, "Remove broken multishop support", merged 2022-03-25. `upgrade/upgrade-5.1.0.php` deletes every `psreassurance_lang` row that does not belong to `PS_SHOP_DEFAULT` and drops `id_shop` from both `psreassurance` and `psreassurance_lang`. It was declared a BC break and it closed #26991, #26979, #26980 and #26531.

The reasons given there are worth repeating, because they are the reasons this is not a small change. MatShir, 2022-01-13: only editing the text of a field ever worked; enabling or disabling an item for a given shop, changing the display and changing the images did not. Core PR #25913 then made the broken support throw. And kpodemski, 2022-03-20, on redoing it: "it's something I want to do in Q2 with a refacto of the module, so it would be a bigger project".

What has changed since is the storage. The module is on 6.0.0 and has moved to Doctrine: `src/Entity/Psreassurance` with a `PsreassuranceRepository` extending `ServiceEntityRepository`, and `classes/ReassuranceActivity.php` is marked `@deprecated 6.0.0 - migrated to src/Form/PsreassuranceFormDataHandler`. The entity has no shop column, and neither does the install SQL in `blockreassurance.php`. So the reimplementation is now an entity plus schema plus upgrade script plus repository scoping plus form handler, not the `id_shop` column the 2022 code had.

That also bears on PrestaShop/blockreassurance#613, open since 2023, which adds `id_shop` to `classes/ReassuranceActivity.php` and nothing else. That file is the deprecated legacy model, so the change cannot reach the Doctrine entity, the repository or the form handler that the configuration page actually goes through today.
